modules/admin-ui-frontend/app/scripts/shared/partials/modals/embedding-code.html

```diff
@@ -0,0 +1,114 @@
+<section ng-show="open" ng-keyup="keyUp($event)" tabindex="1" class="modal modal-animation ng-hide" id="embedding-code">
+  <header>
+    <a class="fa fa-times close-modal" ng-click="close()"></a>
+    <h2 translate="CONFIRMATIONS.ACTIONS.SHOW.EMBEDDING_CODE"><!-- Confirm --></h2>
+  </header>
+
+  <style>
+  .embedSizeButton {
+    display: inline-block;
+    background-color: #466DC3;
+    border: 5px solid #313135;
+    border-radius: 10px;
+    cursor: pointer;
+    margin-right: 5px;
+    vertical-align: middle;
+    font-size: 14px;
+    text-align: center;
+    color: white;
+  }
+
+  .embedSizeButtonSelected {
+    border: 5px solid #FFE04F;
+  }
+
+  .copyConfirm {
+    display: inline-block;
+    text-align: center;
+    margin: 10px 0px 10px 0px;
+    width: 100%;
+    font-weight: bold;
+    font-size: 1.1em;
+    color: #466DC3;
+  }
+  </style>
+
+  <script>
+    var engageUrl = "<SERVER_URL>";
+
+    $(document).ready(function() {
+      var reply = $.ajax({
+        url: "/api/info/organization/properties",
```

Comment:
isn't there a simpler way to get the engage URL? An user needs at least one of these roles (or a new role ROLE_UI_EVENTS_EMBEDDING_CODE) to access the api

`    <sec:intercept-url pattern="/api/info/organization/properties" method="GET" access="ROLE_ADMIN, ROLE_API" />`

Reply:
Plus, `/api` does not necessarily exist on the same server.
I'm not sure if the information is in the admin interface endpoints, you should get it from `/info/me.json` though. The Media Molule link in the top right corner of the admin interface uses the same information, maybe take a look at where it gets the information from.